### ISSUE TITLE
Weird clippy suggestion

### ISSUE THREAD

**Issue by Snoupix** [rust-lang/rust-clippy#15589]
### Summary

It might just be me but this clippy lint feels wrong.

### Lint Name

clippy::collapsible_if

### Reproducer

POC
```rust
let x: Result<(), ()> = Err(());
if 1 > 2 {
    if let Err(_err) = x {}
}

// Clippy suggests

match 1 > 2 {
    true => if let Err(_err) = x {}
    false => (),
}
```

Now that if-let-chains are [stabilized](https://github.com/rust-lang/rust/pull/132833), it should suggest something like this:

```rust
if 1 > 2 && let Err(_err) = x {}
```

IMO For older rust versions, it souldn't suggest anything if one of the if statements is a `if let`.

### Version

```text
rustc 1.89.0 (29483883e 2025-08-04)
binary: rustc
commit-hash: 29483883eed69d5fb4db01964cdf2af4d86e9cb2
commit-date: 2025-08-04
host: x86_64-unknown-linux-gnu
release: 1.89.0
LLVM version: 20.1.7
```

**Comment by ada4a** [comment]
Hm, can't reproduce this:
- on Edition 2021, this doesn't raise a warning either in 1.89.0, 1.90.0-beta.7, or 1.90.0-nightly (2025-08-29)
- on Edition 2024, this gives exactly the suggestion that you propose, on all the aforementioned versions: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=bf4cb7ac125e50a938de0f201d0c61be

**Comment by Snoupix** [comment]
Hi @ada4a , thanks for your feedback.

Running `cargo clippy` gives me what you have on the playground (edition 2024), which right. My fault for not checking.
My issue is really based on the Code Action that rust-analyzer gives me. It only gives "Replace if with match". But it's probably a LSP misconfiguration on my part.

Given that the `cargo clippy` output is right, if your LSP suggests the right fix, this issue can be closed.

**Comment by ada4a** [comment]
Hi @Snoupix, thanks for the quick response:)

Just checked, and I indeed also get such a Code Action. But I don't think that's a misconfiguration.

There are actually two kinds of Code Actions:
- those marked with a yellow bulb in VSCode are indeed fixes for compiler warnings
- but there are also those marked with a blue bulb in VSCode, which are just some common refactorings one might want to do, e.g., when restructuring the code. The "Replace if with match" code action belongs to this category.

You can read more about Code Actions here:
- protocol specification: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_codeAction
- the list of Code Actions available in rust-analyzer: https://rust-analyzer.github.io/book/assists.html

**Comment by Snoupix** [comment]
Quick response time goes both ways !

Oh interesting ! I don't use VSCode but Neovim so that's why it was most likely a misconfiguration on my part.

That's good to know, I always thought that code actions from rust-analyzer's language server were "tied" to clippy suggestions since most of the time you have the same fix suggested.
So yes, I understand my confusion better now, thanks for pointing it out ! :)
